hnn_core/mpi_child.py

```diff
@@ -78,25 +78,21 @@ def run_mpi_simulation():
 
     # send results to stderr
     if rank == 0:
-        # send back dpls and spikedata
-        pickled_string = pickle.dumps(sim_data)
-
-        # pad data before encoding, always add at least 4 "=" to mark end
-        padding = len(pickled_string) % 4
-        pickled_string += b"=" * padding
-        pickled_string += b"=" * 4
-
-        # encode as base64 before sending to stderr
-        repickled_bytes = codecs.encode(pickled_string,
-                                        'base64')
-
         data_iostream = io.BytesIO()
 
         # Force the use of bytes streams under Python 3
         if hasattr(data_iostream, 'buffer'):
             data_iostream = data_iostream.buffer
 
-        data_iostream.write(repickled_bytes)
+        if len(sim_data) > 0:
```

Comment:
The less we mix MPI code with the rest, the better I would say. Most contributors will not need to worry too much about the MPI heavy lifting then ...

Reply:
Is there something specific you are suggesting here?

Reply:
umm ... I am trying to understand/think what this if condition does and if it depends on the result somehow? When I last looked at the code, I was thinking that [this](https://github.com/jonescompneurolab/hnn-core/pull/185/files#diff-ca5f146e027d9dc6ef29ff4db9c21ca2R66-R77) could probably be logically separated from the pickling and process communication etc. into a separate function?

Reply:
It helps the code in `parallal_backends.py` if this only returns data if there's something there. If you pickle an empty string you still get the bytes below. In my troubleshooting, I wanted to throw an exception if the process didn't return anything. Its reasonable to alterntively unpack it and make that determination it was an empty list too.
```
>>> import pickle
>>> string = pickle.dumps('')
>>> import codecs
>>> codecs.encode(string,'base64')
b'gANYAAAAAHEALg==\n'
```

Reply:
It makes sense to remove this if statement now.